File: sort/164.go
import "sort"
func maximumGap(nums []int) int {
    if len(nums) < 2 {
        return 0
    }
    sort.IntSlice(nums).Sort()
    var gap int
    for i := 0; i < len(nums) - 1; i++ {
        if nums[i+1] - nums[i] > gap {
            gap = nums[i+1] - nums[i]
        }
    }
    return gap
}

File: sort/164.cc
class Solution {
public:
    int maximumGap(vector<int>& nums) {
        if (nums.size() < 2)
            return 0;
        std::sort(nums.begin(), nums.end());
        int ret = 0;
        for (int i = 1; i < nums.size(); ++i) {
            ret = std::max(ret, nums[i]-nums[i-1]);
        }
        return ret;
    }
};
